File: 167. Two Sum II - Input array is sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l, r = 0, len(numbers)-1
        while l < r:
            if numbers[l] + numbers[r] ==  target:
                return [l+1, r+1]
            if numbers[l] + numbers[r] > target:
                while r > 0 and numbers[r-1] == numbers[r]:
                    r -= 1
                r -= 1
            else:
                while l < len(numbers)-1 and numbers[l] == numbers[l+1]:
                    l += 1
                l += 1

File: 167. Two Sum II - Input array is sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int lo=0, hi=numbers.length-1;
        while (lo<hi) {
            int temp = numbers[lo] + numbers[hi];
            if (temp == target) {
                return new int[] {lo+1, hi+1};
            }
            if (temp > target) {
                hi--;
            } else {
                lo++;
            }
        }
        return new int[] {0,0};
    }
}